modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,241 @@
+import {Meta, ArgsTable} from '@storybook/addon-docs';
+import {Box} from '@workday/canvas-kit-react/layout';
+import {useResizeObserver, useResponsiveContainerStyles} from '@workday/canvas-kit-react/common';
+import {ResponsiveContainer} from './examples/ResponsiveStyles';
+import {ResponsiveViewport} from './examples/ResponsiveViewport';
+
+<Meta title="Hooks and Utilities/Responsive Styling" />
+
+# Responsive Styling
+
+Responsive styling is an approach to web development that adjusts layout and appearance based on the
+device's viewport size or an element's container size.
+
+## What is Responsive Styling?
+
+Responsive Styling it is a term used to describe an approach to web design or a set of best
+practices, used to create a layout that can _respond_ to the device being used to view the content.
+
+Read more about the history of
+[Responsive Design](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Responsive_Design).
+
+### Container vs Viewport
+
+There are two ways we will look at responsive styling below. They are container-based and
+viewport-based responsive styling.
+
+> Using container-based responsive styling, we will measure the size of the container the element is
+> in and apply styles based on the size of that container.
+
+> Using viewport-based responsive styling, we will measure the size of the user's viewport and apply
+> styles based on the size of that viewport.
+
+If you are measuring a user's viewport size, then you should be using viewport-based responsive
+styling.
+
+If you only want to measure the size of a container, then you should be using container-based
+responsive styling.
+
+> **Note**: A container's size can be affected by surrounding containers or the viewport size.
+
+### Canvas Breakpoints
+
+In Canvas Kit, there are a set of breakpoints:
+
+| Zero  | Small   | Medium  | Large    | Extra Large |
+| ----- | ------- | ------- | -------- | ----------- |
+| > 0px | > 320px | > 768px | > 1024px | > 1440px    |
+
+> Zero: This is applied from 0 - 319px
+
+> Small: This is applied from 320px - 767px
+
+> Medium: This is applied from 768px - 1023px
+
+> Large: This is applied from 1024px - 1439px
+
+> Extra Large: This is applied from 1440px and up
+
+These are the breakpoints we will use when we apply styles based on the size of a container or
+viewport.
```

Comment:
I updated this section to remove more duplicate content and to provide an explanation of breakpoints.
```suggestion
Breakpoints are the points at which the styles will change to provide an optimal user experience.
Think of a container element's or viewport's width as a range of possible sizes broken down into
buckets (e.g., `0-319px`, `320-767px`, `768-1023px`, etc).

Canvas uses the following breakpoints for both container-based and viewport-based responsive
styling:

- `zero`: styles applied from `0px` up until (but not including) `320px`
- `s`: styles applied from `320px` up until (but not including) `768px`
- `m`: styles applied from `768px` up until (but not including) `1024px`
- `l`: styles applied from `1024px` up until (but not including) `1440px`
- `xl`: styles applied from `1440px` and up
```